### DISCUSSION TITLE
mutual information

### DISCUSSION THREAD

**Discussion by buligar** [brainets/frites#65] [Q&A]
Good afternoon, I have a question about the calculation of mutual information. I would like to calculate the mutual information between two populations of neurons. One population consists of an array (number of neurons, membrane potential, time) What are x and y? What should the x and y data look like? Instead of x there should be one population of neurons, and instead of y there should be another population? Or should these two populations be in x? How should I convert vm1_res, vm2_res?
     
vm1_res = vm1.reshape(999,100)
vm2_res = vm2.reshape(999,100)
dataset = DatasetEphy(x,y=y, roi=ch, times=times)
wf = WfMi(mi_type='cc',inference='ffx', verbose=False)
mi, _ = wf.fit(dataset)
plt.figure()
time = np.arange(0.999)
plt.plot(time, mi)
plt.xlabel("Time (s)"), plt.ylabel("MI (bits)")
plt.title('I(C; C)')
plt.show()

**Comment by EtienneCmb** [comment]
Hi @buligar,

The MI encapsulates the statistical dependencies between two variables, just as Pearson or Spearman correlations. I think you're confused here because the `DatasetEphy` is made to relate neural data with behavioral or task-variables (stim, outcomes, reaction times etc.). If you just want to estimate the mount of information between your two populations, then you can use the MI metrics directly from `frites.core` :

```python
from frites.core import gcmi_nd_cc

# pop_1 = (n_neurons, n_potential, n_times)
# pop_2 = (n_neurons, n_potential, n_times)

# compute mi across n_neurons dimension (axis=0)
mi = gcmi_nd_cc(pop_1, pop_2, traxis=0)
# mi.shape = (n_potential, n_times)
```

Hope it is more clear like that,

Best,

**Reply by buligar** [reply]
Are the results correct?
![image](https://github.com/brainets/frites/assets/75439313/aabc96f1-24e3-4615-849e-604a249a1612)
![image](https://github.com/brainets/frites/assets/75439313/83eeeb68-8fca-4dbb-a041-f3ad1e4345cc)

**Comment by EtienneCmb** [comment]
Hard to answer since I don't know the data, the ground truth etc.

**Reply by buligar** [reply]
[vm1all.csv](https://github.com/brainets/frites/files/12845552/vm1all.csv)
[vm2all.csv](https://github.com/brainets/frites/files/12845553/vm2all.csv)
import numpy as np
import pandas as pd
from frites.core import gcmi_nd_cc
import matplotlib.pyplot as plt

df_vm1 = pd.read_csv('vm1all.csv')
df_vm2 = pd.read_csv('vm2all.csv')
vm1all = df_vm1.values
vm2all = df_vm2.values

dt = 1
t = np.array([1, 1000])
tvec = np.arange(t[0], t[1], dt)
plt.figure()
plt.plot(tvec, vm1all.T)
plt.show()
plt.figure()
plt.plot(tvec, vm2all.T)
plt.show()

mi = gcmi_nd_cc(vm1all, vm2all, traxis=0)

plt.figure()
time = np.arange(0, 999)
plt.plot(time, mi)
plt.xlabel("Time (s)"), plt.ylabel("MI (bits)")
plt.title('I(C; D)')
plt.show()
